framework/src/test/java/viewpointtest/ViewpointTestVisitor.java

```diff
@@ -0,0 +1,28 @@
+package viewpointtest;
+
+import com.sun.source.tree.NewClassTree;
+
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/** The visitor for the viewpointtest type-system. */
+public class ViewpointTestVisitor extends BaseTypeVisitor<ViewpointTestAnnotatedTypeFactory> {
+    /**
+     * Create a new ViewpointTestVisitor.
+     *
+     * @param checker the checker to which this visitor belongs
+     */
+    public ViewpointTestVisitor(BaseTypeChecker checker) {
+        super(checker);
+    }
+
+    @Override
+    public Void visitNewClass(NewClassTree tree, Void p) {
+        AnnotatedTypeMirror Type = atypeFactory.getAnnotatedType(tree);
+        if (Type.hasAnnotation(atypeFactory.TOP)) {
```

Comment:
Shouldn't `LOST` also be forbidden?

Reply:
Yes, how about `poly` and `RD` qualifier. I left out those two as I feel like we need more discussion.